odl/solvers/nonsmooth/douglas_rachford.py

```diff
@@ -224,3 +244,61 @@ def douglas_rachford_pd(x, f, g, L, tau, sigma, niter,
     # The final result is actually in p1 according to the algorithm, so we need
     # to assign here.
     x.assign(p1)
+
+
+def douglas_rachford_pd_stepsize(L, tau=None):
+    r"""Compute defautl step sizes for the Douglas-Rachford solver.
+
+    Parameters
+    ----------
+    L : sequence of `Operator` or float
+        The operators or the norms of the operators that are used in the
+        `douglas_rachford_pd` method. For `Operator` entries, the norm
+        is computed with `power_method_opnorm`.
+    tau : positive float, optional
+        Use this value for ``tau`` instead of computing it from the
+        operator norms, see Notes.
+
+    Returns
+    -------
+    tau : float
+        The ``tau`` step size parameter for the primal update.
+    sigma : tuple of float
+        The ``sigma`` step size parameters for the dual update.
+
+    Notes
+    -----
+    To guarantee convergence, the parameters :math:`\tau`, :math:`\sigma_i`
+    and :math:`L_i` need to satisfy
+
+    .. math::
+       \tau \sum_{i=1}^n \sigma_i \|L_i\|^2 < 4.
+
+    An example of such a choice is
+
+    .. math::
+        \tau = \frac{1}{\sum_{i=1}^n \|L_i\|},
+        \quad
+        \sigma = \frac{2}{n \tau \|L_i\|^2}
+
+    which are the defaults for this implementation.
+    """
+    L_norms = []
+    for Li in L:
+        if np.isscalar(Li):
+            L_norms.append(float(Li))
+        elif isinstance(Li, Operator):
+            L_norms.append(Li.norm(estimate=True))
+        else:
+            raise TypeError('invalid entry {!r} in `L`'.format(Li))
+
+    if tau is None:
+        norm_sum = sum(L_norms)
+        tau = 1 / norm_sum if norm_sum != 0 else 1.0
```

Comment:
Frankly I'd prefer throwing an error in this case. Obviously something went wrong, no?

Reply:
I guess so. I was just thinking a bit back and forth about how to handle operators that don't matter, which are those that are used with an indicator function. There you can take whatever step size you like and only take the rest of the ingredients into account. But it seems to be easiest to do that manually. So let's raise if there is a nonpositive norm anywhere.